test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/retry/ExceptionAttemptMessageBehaviorTest.java

```diff
@@ -0,0 +1,166 @@
+import software.amazon.awssdk.awscore.client.builder.AwsClientBuilder;
+import software.amazon.awssdk.awscore.exception.AwsServiceException;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.protocolrestjson.ProtocolRestJsonAsyncClient;
+import software.amazon.awssdk.services.protocolrestjson.ProtocolRestJsonAsyncClientBuilder;
+import software.amazon.awssdk.services.protocolrestjson.ProtocolRestJsonClient;
+import software.amazon.awssdk.services.protocolrestjson.ProtocolRestJsonClientBuilder;
+import software.amazon.awssdk.services.protocolrestjson.model.AllTypesRequest;
+import software.amazon.awssdk.services.protocolrestjson.model.AllTypesResponse;
+
+
+/**
+ * A set of tests that verify the behavior of the SDK when attempts are added to the exception message.
+ */
+public abstract class ExceptionAttemptMessageBehaviorTest<ClientT, BuilderT extends AwsClientBuilder<BuilderT, ClientT>> {
+
+    protected WireMockServer wireMock = new WireMockServer(0);
+
+    protected abstract BuilderT newClientBuilder();
+
+    protected abstract AllTypesResponse callAllTypes(ClientT client);
+
+    private BuilderT clientBuilder() {
+        StaticCredentialsProvider credentialsProvider =
+            StaticCredentialsProvider.create(AwsBasicCredentials.create("akid", "skid"));
+        return newClientBuilder()
+            .credentialsProvider(credentialsProvider)
+            .region(Region.US_EAST_1)
+            .endpointOverride(URI.create("http://localhost:" + wireMock.port()));
+    }
+
+    @BeforeEach
+    public void beforeEach() {
+        wireMock.start();
+    }
+
+    @AfterEach
+    public void afterEach() {
+        wireMock.stop();
+    }
+
+    @Test
+    public void exceptionMessage_whenRetryable_includesMultipleAttempts() {
+        wireMock.stubFor(post(anyUrl()).willReturn(aResponse().withStatus(429)));
+
+        ClientT client = clientBuilder()
+            .overrideConfiguration(o -> o.retryStrategy(
+                b -> b.retryOnException(AwsServiceException.class)))
+            .build();
+
+        AwsServiceException exception = assertThrows(AwsServiceException.class,
+                                                     () -> callAllTypes(client));
+
+        assertThat(exception.getMessage()).contains("(Attempts: 4)");
+        wireMock.verify(4, postRequestedFor(anyUrl()));
+    }
+
+    @Test
+    public void exceptionMessage_whenNonRetryable_includesSingleAttempt() {
+        wireMock.stubFor(post(anyUrl()).willReturn(aResponse().withStatus(403)));
+
+        ClientT client = clientBuilder()
+            .overrideConfiguration(o -> o.retryStrategy(
+                b -> b.retryOnException(AwsServiceException.class)))
+            .build();
+
+        AwsServiceException exception = assertThrows(AwsServiceException.class,
+                                                     () -> callAllTypes(client));
+
+        assertThat(exception.getMessage()).contains("(Attempts: 1");
```

Comment:
Can we `"(Attempts: 1)"` since this test case will also pass if we out put as "(Attempts: 10" or "(Attempts: 1__SPACE_" etc

Reply:
good catch! Ill fix that.